### PR TITLE
Add reporter

### LINKED ISSUE #238: Add `teal.reporter` into tmh modules
- blocked by https://github.com/insightsengineering/teal.modules.clinical/issues/505
After https://github.com/insightsengineering/teal/pull/635 teal applications can be used to generate reports - do we want them added into tmh as well?

### PR DESCRIPTION
**Pull request by mhallal1**
closes #238 

Add `reporter` to all modules including:
1) plot/table
2) all selected encodings
3) filter panel

the PCA correlation plot depends on this PR: https://github.com/insightsengineering/teal.reporter/pull/113

### CODE REVIEW

**Comment by Polkas** [comment]
Modules combined `teal.gallery::launch_app("RNA-seq")`

Modules separately: 

 - [ ] tm_g_barplot
 - [ ] tm_g_boxplot
 - [ ] tm_g_forest_tte
 - [ ] tm_g_km
 - [ ] tm_g_pca
 - [ ] tm_g_quality
 - [ ] tm_g_scatterplot
 - [ ] tm_g_volcanoplot

File: R/forestplot.R [hunk 1]
```diff
@@ -210,12 +220,56 @@ srv_g_forest_tte <- function(id,
...
+    ### REPORTER
+    if (with_reporter) {
+      card_fun <- function(comment) {
+        card <- teal.reporter::TealReportCard$new()
+        card$set_name("Boxplot")
+        card$append_text("Boxplot", "header2")
+        card$append_text("Filter State", "header3")
+        card$append_fs(datasets$get_filter_state())
+        card$append_text("Selected Options", "header3")
+        browser()
```
Comment (Polkas) [inline R/forestplot.R:239]:
```suggestion
        browser()
```

Please rm it:)

Comment (Polkas) [inline R/forestplot.R:239]:
```suggestion
        browser()
```

File: R/forestplot.R [hunk 2]
```diff
@@ -210,12 +220,55 @@ srv_g_forest_tte <- function(id,
...
       height = plot_height,
       width = plot_width
     )
+
+    ### REPORTER
+    if (with_reporter) {
+      card_fun <- function(comment) {
+        card <- teal.reporter::TealReportCard$new()
+        card$set_name("Boxplot")
+        card$append_text("Boxplot", "header2")
```
Comment (Polkas) [inline R/forestplot.R:235]:
```suggestion
        card$set_name("Forest Plot")
        card$append_text("Forest Plot", "header2")
```

**Comment by Polkas** [comment]
It is surprising that only one module is using plot_with_settings/table_with_settings.

**Comment by mhallal1** [comment]
> Please regenarate or remove the _snaps directory in the testthat. We have some Errors under the testing depth 5 nevertheless they are already on the main branch.

Issue created as it existed on main already: https://github.com/insightsengineering/teal.modules.hermes/issues/266

**Comment by mhallal1** [comment]
@danielinteractive we had to update the `snaps` folder to pass the test with a depth of 5:
Screenshot from 2022-07-13 11-37-45

**Comment by mhallal1** [comment]
> Please do not print the Encoding elements if they are optional and empty. image

fixed.

**Review by Polkas — APPROVED** [review]
Great work:)
Please create an issue for checking the plot with settings (empty returned height and width) in forest plot, thanks.

File: R/forestplot.R [hunk 3]
```diff
@@ -210,12 +220,55 @@ srv_g_forest_tte <- function(id,
...
+            input$`adtte-paramcd`,
+            "\nProbability Cutoff:",
+            input$probs,
+            "\nSubgroup Variable:",
+            input$`subgroups-sample_var`
+          ),
+          style = "verbatim"
+        )
+        card$append_text("Plot", "header3")
+        card$append_plot(forest_plot())
```
[the remark this answers is not in the document]
Comment (mhallal1) [reply R/forestplot.R]: issue created: https://github.com/insightsengineering/teal.modules.hermes/issues/267

**Pull request merged**